docs/source/customizations/rhai.mdx

```diff
@@ -1,348 +1,191 @@
 ---
 title: Rhai scripts for the Apollo Router
-description: Extend router functionality without compiling a custom plugin
+description: Extend your router without building a custom binary
 ---
 
-The Apollo Router provides support for scripts that use the [Rhai](http://rhai.rs) scripting language. Rhai is useful for performing common script-based tasks (manipulating strings, processing headers, etc.) in a Rust-based project. If you want to learn more about simple programming with Rhai, the [book](https://rhai.rs/book/) has a [language reference](https://rhai.rs/book/language/) which is very helpful.
+You can customize the Apollo Router's behavior with scripts that use the [Rhai scripting language](https://rhai.rs/book/). Rhai is ideal for performing common scripting tasks (manipulating strings, processing headers, etc.) in a Rust-based project.
 
-Rhai scripts hook into the same [Apollo Router lifecycle events](./overview/#how-customizations-work) as native Rust plugins.
+> To start familiarizing yourself with Rhai, see the [language reference](https://rhai.rs/book/language/) and some basic [syntax examples](https://rhai.rs/book/start/examples/scripts.html).
 
-## Limitations
+Rhai scripts can hook into any combination of services in the Apollo Router's [request-handling pipeline](./overview/#how-customizations-work).
 
-Like the rest of the Apollo Router, Rhai support is in active development. At this time, functionality is limited.
+## Use cases
 
-**What you can do:**
-* Manipulate request/response HTTP headers
-* Manipulate request/response context
-* Perform `checkpoint`-style short circuiting of requests
-* Modify the status codes of requests/responses
-* Modify the body of requests
-* Modify the body of responses
+Common use cases for Apollo Router Rhai scripts include:
 
-**What you can't do (yet):**
-* Execute calls to external services
+* Modifying the details of HTTP requests and responses. This includes requests sent from clients to your router, along with requests sent from your router to your subgraphs. You can modify any combination of the following:
```

Comment:
Consider adding logging as a common use case. Many customer customisations can be summarised as "logging some information"...